### ISSUE TITLE
non-integer indexing in Plexon.IO

### ISSUE THREAD

**Issue by Palaeocortex** [NeuralEnsemble/python-neo#211]
Hi group!

I am trying to import Plexon files and I keep getting this message in the terminal, without anything being loaded into the db:

  sigarrays[chan][sample_positions[chan] : sample_positions[chan]+data.size] = data
/usr/local/lib/python2.7/dist-packages/neo/io/plexonio.py:233: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future

This message keeps appearing continuously. After 30 min I end OpenElectrophy and nothing is in the db. My files contain 1-2h long recordings and have 100 - 200 MB (low sampling rate LFP signals).

It looks like the non-integer indexing in scikit-learn will be removed in future versions, so plexon.io needs an update:

https://github.com/numpy/numpy/pull/3243
http://stackoverflow.com/questions/20084218/deprecation-warning-in-scikit-learn-svmlight-format-loader

The continuous terminal output makes loading impossible. One can start openElectrophy with the &>  /dev/null command, which works well but no error messages appears making trouble shooting impossible. 

Basically it is not possible to load .plx data with neo.io.

**Comment by pmeier82** [comment]
There is a new version of the .plx format (1.07 in my case) that is not
handled well by the current IO. I think it has to do with the a change in
the header bitfields, that can handle channel counts >255 now.

Are you still taking pull requests for neo 0.3x? I have working code to
read current Plexon and Blackrock files, but need to squeeze it into IO
classes.

--
φιλ

**Comment by Palaeocortex** [comment]
I think it is rather related to the indexing of arrays than the channel counts. The index var has to be int not float. At least that is what I understand from the error message and from the links I posted.

PlexUtil shows Version 106 and Acquiring software OmniPlex 1.4.3.

**Comment by apdavison** [comment]
> Are you still taking pull requests for neo 0.3x?

Yes!

**Comment by pmeier82** [comment]
Aye, that is a quite common warning (DeprecationWarning) and not an error.
The only practical impact should be the flooding of the console. You can
ignore warnings like so:
http://stackoverflow.com/questions/14463277/how-to-disable-python-warnings.

I guess this has to be addressed on the neo.io level, perhaps by
refactoring the indexing on the base class level. Seems to be stem from the
use of quantities and their conversion to array indices.


​Still, this might not be your actual problem. I had problems reading .plx
file with versions >1.04 with the 0.3x branch.

--
φιλ

**Comment by pmeier82** [comment]
Just discovered that there are updated versions PlexonIO and BlackrockIO
(and several others) on the master branch. I have to check in detail when I
find the time, but I think they already cover what I could contribute.

@Palaeocortex
​Not sure what version of neo you are using, if you got it by
pip/easyinstall or the like it should be the 0.3.3 release with the old io
classes. In that case switching to the master branch might already solve
your problem.


--
φιλ

**Comment by samuelgarcia** [comment]
Hi.
Yes you should take the master branch. It is still compatible with .3.x schema but not for long time!
If you want push recent comit of PlexonIO to 0.3.x send a PR.

Note that the plx format is pretty old and its internal strucutre make it impossible to parse it fastly.
I spend hours on improving the speed with no succes. Each channel is coded in variable chunks at variable positions. So you can't anything in pure python/numpy because need loops. You need some C code for that and even with C codes it should be slow. Plexon have design some years ago a new format but unsupported by neo.

**Comment by Palaeocortex** [comment]
Thanks to all of you for the lightspeed reactions! I am able to load data by suppressing the warnings. They are just warnings but the problem is that warning output to the console slows down the program. Without warnings I load the file in 5 min, with warnings it is not ready after 30 min. Thanks also for the idea of suppressing just the warnings, until now I started OE with &> /dev/null suppressing all output. Yet this is not the best solution because I need debugging.

**Comment by samuelgarcia** [comment]
This concern old branches. So close.
